src/utils/response.ts

```diff
@@ -107,10 +107,16 @@ export function getResponseHeader(
 
 export function setResponseHeaders(
   event: H3Event,
-  headers: Record<HTTPHeaderName, Parameters<OutgoingMessage["setHeader"]>[1]>,
+  headers: Partial<
+    Record<HTTPHeaderName, Parameters<OutgoingMessage["setHeader"]>[1]>
+  >,
 ): void {
   for (const [name, value] of Object.entries(headers)) {
-    event.node.res.setHeader(name, value);
+    if (value === undefined) {
```

Comment:
I do not think we need to add logic to cover this. I think we can just assert `value!`.

Reply:
changed :raised_hands:

Reply:
for reference Node.js will throw `ERR_HTTP_INVALID_HEADER_VALUE` for undefined https://github.com/nodejs/node/blob/main/lib/_http_outgoing.js#L677